lib/types.ts

```diff
@@ -0,0 +1,85 @@
+import type * as typesObject from './constant/types'
+import type {ValueOf} from 'type-fest'
+
+export interface Point {
+  line: number
+  column: number
+  offset: number
+  _index: number
+  _bufferIndex: number
+}
+export interface Token {
+  start: Point
+  end: Point
+  type: unknown
+  contentType: string
+  previous: Token
+  next: Token
+  _size: number
+  _break?: boolean
+  _tokenizer?: unknown
+  _events?: Event[]
+  _subevents?: Event[]
+  _contentTokenized?: unknown
+  _close?: unknown
+  _marker?: unknown
+  _open?: unknown
+  _side?: number
+  _tabs?: unknown
```

Comment:
Generally, tabs and spaces behave the same, but in the case of a hard break through trailing spaces (`  \n`), tabs don’t work.

Reply:
would this be `number | 'tab'` similar to remark?

Reply:
I think it’s an optional boolean

Reply:
(describes whether a token includes tabs, only used for trailing whitespace: https://github.com/micromark/micromark/blob/66f33dd1d3c22f7ed8ff0d12d38dd8cc5cba4165/lib/tokenize/whitespace.js)